myWork/lib/readFile.js

```diff
@@ -0,0 +1,81 @@
+//       console.log('content:', `one.txt ${data.toString('hex', 0, 8)}`);
+//       let result = `${data.toString('hex', 0, 8)} `;
+//
+//       return fs.readFile(`${__dirname}/../data/two.txt`, function(err, data) {
+//         if(err) throw err('error reading data file two.txt');
+//           if(err) throw err();
+//           console.log(`two.txt ${data.toString('hex', 0, 8)}`);
+//           result+=`${data.toString('hex', 0, 8)} `;
+//
+//             return fs.readFile(`${__dirname}/../data/three.txt`, function(err, data) {
+//               if(err) throw err('error reading data file three.txt');
+//               console.log(`three.txt ${data.toString('hex', 0, 8)}`);
+//               result+=data.toString('hex', 0, 8);
+//               console.log(result);
+//               return result;
+//             });
+//         });
+//     });
+// };
+
+
+// module.exports = function(callback) {
+//   fs.readFile(`${__dirname}/../data/one.txt`, function(err, data) {
+//       if(err) throw err('error reading data file one.txt');
+//       console.log('content:', `one.txt ${data.toString('hex', 0, 8)}`);
+//       let result = `${data.toString('hex', 0, 8)}`;
+//
+//       return fs.readFile(`${__dirname}/../data/two.txt`, function(err, data) {
+//         if(err) throw err('error reading data file two.txt');
+//           if(err) throw err();
+//           console.log(`two.txt ${data.toString('hex', 0, 8)}`);
+//           result+=`${data.toString('hex', 0, 8)}`;
+//
+//             return fs.readFile(`${__dirname}/../data/three.txt`, function(err, data) {
+//               if(err) throw err('error reading data file three.txt');
+//               console.log(`three.txt ${data.toString('hex', 0, 8)}`);
+//               result+=data.toString('hex', 0, 8);
+//               console.log(result);
+//               callback(result);
+//             });
+//         });
+//     });
+// };
+
+
+module.exports = exports = {};
+
+exports.readFile = function(callback) {
+  fs.readFile(`${__dirname}/../data/one.txt`, function(err, data) {
+      if(err) throw err('error reading data file one.txt');
+      console.log('content:', `one.txt ${data.toString('hex', 0, 8)}`);
+      let result = `${data.toString('hex', 0, 8)} `;
+
+      return fs.readFile(`${__dirname}/../data/two.txt`, function(err, data) {
+        if(err) throw err('error reading data file two.txt');
+          if(err) throw err();
+          console.log(`two.txt ${data.toString('hex', 0, 8)}`);
+          result+=`${data.toString('hex', 0, 8)} `;
+
+            return fs.readFile(`${__dirname}/../data/three.txt`, function(err, data) {
+              if(err) throw err('error reading data file three.txt');
+              console.log(`three.txt ${data.toString('hex', 0, 8)}`);
+              result+=data.toString('hex', 0, 8);
+              console.log(result);
+              callback(result);
+            });
+        });
+    });
+};
+
+exports.myCallback = function(result) {
```

Comment:
Your callback should be put inline in your tests. It makes it easier to tell where the data is coming from and will help with async testing. Look at line 9 in your tests.

Also you should check in on the arguments of your callback. Remember it should be `callback(err, data)`. Then when errors happen (like line 68), you should pass the error back through the callback (`callback(err);`).